src/d_main.c

```diff
@@ -571,6 +573,81 @@ void D_StartTitle (void)
 // print title for every printed line
 static char title[128];
 
+char **tempdirs = NULL;
+
+static void AutoLoadWADs(const char *path);
+
+static boolean D_AddZipFile(const char *file)
+{
+  int i;
+  mz_zip_archive zip_archive;
+  char *str, *tempdir;
+  static int idx = 0;
+
+  str = M_StringDuplicate(file);
+  M_ForceLowercase(str);
+  if (!M_StringEndsWith(str, ".zip"))
+  {
+    free(str);
+    return false;
+  }
+  free(str);
+
+  memset(&zip_archive, 0, sizeof(zip_archive));
+  if (!mz_zip_reader_init_file(&zip_archive, file, MZ_ZIP_FLAG_DO_NOT_SORT_CENTRAL_DIRECTORY))
+  {
+    printf("D_AddZipFile: Failed to open %s\n", file);
+    return true;
+  }
+
+  str = M_StringJoin("_", PROJECT_SHORTNAME, "_", M_BaseName(file), NULL);
+  tempdir = M_TempFile(str);
+  free(str);
+  M_MakeDirectory(tempdir);
+
+  for (i = 0; i < (int)mz_zip_reader_get_num_files(&zip_archive); ++i)
+  {
+    mz_zip_archive_file_stat file_stat;
+    char *name;
+
+    mz_zip_reader_file_stat(&zip_archive, i, &file_stat);
+
+    if (file_stat.m_is_directory)
+      continue;
+
+    name = M_StringDuplicate(M_BaseName(file_stat.m_filename));
+    M_ForceLowercase(name);
```

Comment:
Should we probably introduce a case-insensitive variant of `M_StringEndsWith()`?

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/649/commits/4a8b277a56c7c9e0fb0d9a0ee57ba26958ba1ef5